src/mercury_engine_data_structures/formats/bcskla.py

```diff
@@ -22,50 +20,47 @@
     this,
 )
 
-from mercury_engine_data_structures.adapters.enum_adapter import EnumAdapter
 from mercury_engine_data_structures.common_types import Float, VersionAdapter
 from mercury_engine_data_structures.construct_extensions.alignment import AlignTo
 from mercury_engine_data_structures.formats.base_resource import BaseResource
 from mercury_engine_data_structures.formats.property_enum import PropertyEnumDoubleUnsafe
 from mercury_engine_data_structures.game_check import Game
 
 
-class TimingTypeEnum(Enum):
-    ONE_BYTE = 8
-    TWO_BYTE = 0
-
-
 class DreadKFVAdapter(Adapter):
+    class TimingType:
+        ONE_BYTE = 8
+        TWO_BYTE = 0
+
     SUBCON = Struct(
-        timing_type=EnumAdapter(TimingTypeEnum, Int16ul),
+        timing_type=Int16ul,
         count=Int16ul,
-        timings=IfThenElse(this.timing_type == TimingTypeEnum.TWO_BYTE, Int16ul[this.count], Int8ul[this.count]),
+        timings=IfThenElse(this.timing_type == TimingType.TWO_BYTE, Int16ul[this.count], Int8ul[this.count]),
         _padding=AlignTo(4, b"\xff"),
-        values=Array(this.count, Struct(value=Float, derivative=Float)),
-    )
+        values=Array(this.count, Float[2]),
+    ).compile()
 
     def __init__(self):
         super().__init__(self.SUBCON)
 
     def _decode(self, obj, context, path):
-        res = ListContainer()
-        for i in range(obj.count):
-            res.append(
-                Container(
-                    time=obj.timings[i],
-                    value=obj["values"][i].value,
-                    derivative=obj["values"][i].derivative,
-                )
+        return ListContainer(
+            Container(
+                time=time,
+                value=v[0],
+                derivative=v[1],
             )
-
-        return res
+            for time, v in zip(obj.timings, obj["values"], strict=True)
+        )
 
     def _encode(self, obj, context, path):
         res = Container(
-            timing_type=TimingTypeEnum.TWO_BYTE if obj[-1].time > 0xFF else TimingTypeEnum.ONE_BYTE,
+            timing_type=DreadKFVAdapter.TimingType.TWO_BYTE
+            if obj[-1].time > 0xFF
+            else DreadKFVAdapter.TimingType.ONE_BYTE,
             count=len(obj),
             timings=ListContainer([v.time for v in obj]),
-            values=ListContainer([Container(value=v.value, derivative=v.derivative) for v in obj]),
+            values=[[v.value, v.derivative] for v in obj],
```

Comment:
if you're making this a python list instead of ListContainer probably should do the same with timings

Reply:
It's less about being a list vs ListContainer and more about being a list instead of a dict. But I guess it could be a list with both

Reply:
I think it's better to make res use either construct (everything is a Container/ListContainer) or python (everything is a dict/list). doesn't matter for the code but i think it's better in terms of style. I'd be fine with `values=ListContainer([...])` as well.

Reply:
would a NamedTuple instead of a dict work here? it would be nice to keep the names

Reply:
This is only between the encode and the subconstruct. Nothing is different outside of this class. This is a very hot path too, so I want to use whatever's fastest.